src/WorldStorage/TreeNBT.h

```diff
@@ -0,0 +1,484 @@
+	using const_iterator  = cTagArray::const_iterator;
+
+	// No modifiable references can be handed out in order to maintain
+	// the invariant that all tags are of the same type
+	using reference = const_reference;
+	using pointer   = const_pointer;
+	using iterator  = const_iterator;
+
+	/** Try to insert a new tag at the given position.
+	Returns false if the tag is of the wrong type and cannot be inserted. */
+	std::pair<iterator, bool> TryInsert(iterator a_Pos, const cTag & a_Tag)
+	{
+		if (!IsTagCompatible(a_Tag.TypeId()))
+		{
+			return {a_Pos, false};
+		}
+
+		return {m_Tags.insert(MutableIterator(a_Pos), std::move(a_Tag)), true};
+	}
+
+	/** Try to insert a new tag at the given position.
+	Returns false if the tag is of the wrong type and cannot be inserted. */
+	std::pair<iterator, bool> TryInsert(iterator a_Pos, cTag && a_Tag)
+	{
+		if (!IsTagCompatible(a_Tag.TypeId()))
+		{
+			return {a_Pos, false};
+		}
+
+		return {m_Tags.insert(MutableIterator(a_Pos), std::move(a_Tag)), true};
+	}
+
+	/** Try to insert a new tag at the end of the list.
+	Returns false if the tag is of the wrong type and cannot be inserted. */
+	bool TryPushBack(const cTag & a_Tag)
+	{
+		if (!IsTagCompatible(a_Tag.TypeId()))
+		{
+			return false;
+		}
+
+		m_Tags.push_back(a_Tag);
+		return true;
+	}
+
+	/** Try to insert a new tag at the end of the list.
+	Returns false if the tag is of the wrong type and cannot be inserted. */
+	bool TryPushBack(cTag && a_Tag)
+	{
+		if (!IsTagCompatible(a_Tag.TypeId()))
+		{
+			return false;
+		}
+
+		m_Tags.push_back(std::move(a_Tag));
+		return true;
+	}
+
+	/** Visit the element of the list at the given position. */
+	template <typename Func>
+	void Visit(iterator a_Pos, Func && a_Visitor)
+	{
+		MutableIterator(a_Pos)->Visit(std::forward<Func>(a_Visitor));
+	}
+
+	/** Visit the element of the list at the given position. */
+	template <typename Func>
+	void Visit(const_iterator a_Pos, Func && a_Visitor) const
+	{
+		a_Pos->Visit(std::forward<Func>(a_Visitor));
+	}
+
+	/** Type of the current elements or TAG_End if empty. */
+	eTagType TypeId() const
+	{
+		return m_Tags.empty() ? TAG_End : m_Tags[0].TypeId();
+	}
+
+	// Standard STL style interface
+
+	const cTag & operator [] (size_type a_Idx) const
+	{
+		return m_Tags[a_Idx];
+	}
+
+	iterator cbegin() const { return m_Tags.cbegin(); }
+	iterator begin() const { return m_Tags.cbegin(); }
+
+	iterator cend() const { return m_Tags.cend(); }
+	iterator end() const { return m_Tags.cend(); }
+
+
+	iterator erase(iterator a_Pos)
+	{
+		return m_Tags.erase(MutableIterator(a_Pos));
+	}
+
+	iterator erase(iterator a_First, iterator a_Last)
+	{
+		return m_Tags.erase(MutableIterator(a_First), MutableIterator(a_Last));
+	}
+
+	size_type size() const { return m_Tags.size(); }
+	bool empty() const { return m_Tags.empty(); }
+
+	void reserve(size_type a_ReserveSize) { m_Tags.reserve(a_ReserveSize); }
+
+	void clear() { m_Tags.clear(); }
+
+	void swap(cList & a_Other) NOEXCEPT
```

Comment:
```suggestion
	void swap(cList & a_Other) noexcept
```

MSVC doesn't like this. Didn't check other